stories/s2l2a.stories.js

```diff
@@ -843,6 +843,93 @@ export const getMapProcessingGainGammaCheckTransparency = () => {
   return wrapperEl;
 };
 
+export const getMapProcessingAdvancedRGB = () => {
+  const imgOriginal = document.createElement('img');
+  imgOriginal.width = '256';
+  imgOriginal.height = '256';
+
+  const imgRGB1 = document.createElement('img');
+  imgRGB1.width = '256';
+  imgRGB1.height = '256';
+
+  const imgRGB2 = document.createElement('img');
+  imgRGB2.width = '256';
+  imgRGB2.height = '256';
+
+  const wrapperEl = document.createElement('div');
+  wrapperEl.innerHTML = '<h2>S2L2A getMapProcessingAdvancedRGB</h2>';
+  wrapperEl.innerHTML += '<h4>no effects | same value | function</h4>';
+  wrapperEl.insertAdjacentElement('beforeend', imgOriginal);
+  wrapperEl.insertAdjacentElement('beforeend', imgRGB1);
+  wrapperEl.insertAdjacentElement('beforeend', imgRGB2);
+
+  const perform = async () => {
+    await setAuthTokenWithOAuthCredentials();
+
+    const layerS2L2A = new S2L2ALayer({
+      instanceId,
+      layerId,
+      maxCloudCoverPercent: 0,
+      evalscript: `
+      //VERSION=3
+      function setup() {
+        return {
+          input: ["B02", "B03", "B04"],
+          output: { bands: 3 }
+        };
+      }
+
+      function evaluatePixel(sample) {
+        return [2.5 * sample.B04, 2.5 * sample.B03, 2.5 * sample.B02];
+      }
+    `,
+    });
+
+    const getMapParams = {
+      bbox: bbox4326,
+      fromTime: new Date(Date.UTC(2018, 9 - 1, 22, 0, 0, 0)),
+      toTime: new Date(Date.UTC(2018, 12 - 1, 22, 23, 59, 59)),
+      width: 512,
+      height: 512,
+      format: MimeTypes.JPEG,
+    };
+
+    const getMapParamsRGB1 = {
+      ...getMapParams,
+      effects: {
+        customEffect: ({ r, g, b }) => ({ r, g, b }),
+      },
+    };
+
+    const getMapParamsRGB2 = {
+      ...getMapParams,
+      effects: {
+        customEffect: ({ r, g, b }) => ({
+          r: r < 0.4 ? r * 2 : r,
+          g: g < 0.4 ? g * 2 : g,
+          b: b < 0.4 ? b * 2 : b,
```

Comment:
I would suggest using a function which more clearly demonstrates the functionality by giving more predictable results, for example:
```
        customEffect: ({ r, g, b }) => ({
          r: r + g + b < 0.6 ? 0 : 1,
          g: r + g + b < 0.6 ? 0 : 1,
          b: r + g + b < 0.6 ? 0 : 1,
        }),
```

Reply:
Thanks, used suggested function